### DISCUSSION TITLE
STDP cut-off point for pre- and post-synaptic calculations?

### DISCUSSION THREAD

**Discussion by rafaelblevin821** [BindsNET/bindsnet#674] [General]
Hi,

Does anyone know the cut-off point for pre- and post-synaptic spikes in STDP calculations used in the online STDP method in Bindsnet?

**Comment by Hananel-Hazan** [comment]
What do you mean `cut-off point for pre- and post-synaptic spikes `?

**Reply by rafaelblevin821** [reply]
I mean, is there a specific amount of time that the pre- and post-synaptic spike timings are considered for the STDP weight updates?

STDP considers the timing of pre- pand post-synaptic spikes to calculate the weight updates does it not?

Is there a cut-off point? I assume they are not considered over the entire time simulation, that would not work.

Thanks for your help, BTW, I appreciate it.

**Reply by Hananel-Hazan** [reply]
The timing of pre- and postsynaptic events doesn't have a universally agreed-upon definition; various experiments yield different values. My suggestion is to treat it as a variable that you can adjust to find the optimal timing for your needs, while ensuring the correct polarity is maintained. Some models even eliminate one side altogether, or use techniques like Anti-STDP to reverse the polarity (you can do this in BindsNET too by flipping the strength).

We're currently developing a couple of optimization techniques to intelligently fine-tune this process, leveraging BindsNET parameters. However, these techniques aren't yet ready for public release. Stay tuned for updates!
